ambry-api/src/main/java/com/github/ambry/frontend/NamedBlobPath.java

```diff
@@ -90,7 +91,10 @@ public static NamedBlobPath parseS3(String path, Map<String, Object> args) throw
     String[] splitPath = path.split("/", 4);
     String blobNamePrefix = RestUtils.getHeader(args, PREFIX_PARAM, false);
     boolean isGetObjectLockRequest = args.containsKey(OBJECT_LOCK_PARAM);
-    boolean isListRequest = blobNamePrefix != null;
+    //There are two cases for S3 listing
+    //1.has prefix (Ex:GET /?prefix=prefixName&delimiter=&encoding-type=url)
+    //2.no prefix but listObjectV2 (Ex:GET /?list-type=2&prefix=&delimiter=&encoding-type=url)
+    boolean isListRequest = blobNamePrefix != null || "2".equals(RestUtils.getHeader(args, LIST_TYPE, false));
```

Comment:
Please use a constant to replace "2". Something like "LIST_TYPE_VERSION_2".

Reply:
Done